docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+Building image for target platform: org.graalvm.nativeimage.Platform$LINUX_AMD64
+Using native toolchain:
+   Name: GNU project C and C++ compiler (gcc)
+   Vendor: redhat
+   Version: 8.5.0
+   Target architecture: x86_64
+   Path: /usr/bin/gcc
+Using CLibrary: com.oracle.svm.core.posix.linux.libc.GLib
+----
+
+The native library info file contains information on the static libraries added to the binary and the other libraries dynamically linked to the executable:
+
+[source,bash]
+----
+$ cat target/*/reports/native_library_info_*.txt
+Static libraries:
+   ../opt/mandrel/lib/svm/clibraries/linux-amd64/liblibchelper.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libnet.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libextnet.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libnio.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libjava.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libfdlibm.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libsunec.a
+   ../opt/mandrel/lib/static/linux-amd64/glibc/libzip.a
+   ../opt/mandrel/lib/svm/clibraries/linux-amd64/libjvm.a
+Other libraries: stdc++,pthread,dl,z,rt
+----
+
+Even more detail can be obtained by passing in `--verbose` as an additional native-image build argument.
+This option can be very useful in detecting whether the options that you pass at a high level via Quarkus are being passed down to the native executable production,
+or whether some third party jar has some native-image configuration embedded in it that is reaching the native-image invocation:
+
+[source,bash,subs=attributes+]
+----
+./mvnw package -DskipTests -Pnative -Dquarkus.native.container-build=true \
+    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:{mandrel-flavor} \
+    -Dquarkus.native.additional-build-args=--verbose
+----
+
+Running with `--verbose` demonstrates how the native-image building process is 2 sequential java processes:
+
+* The first is a very short java process that does some basic validation and builds the arguments for the second process
+(in a stock GraalVM distribution, this is executed as native code).
+* The second Java process is where the meat of the native executable production happens.
+The `--verbose` option shows the actual Java process executed.
+You could take the output and run it yourself.
+
+One may also combine multiple native-image build options by separating with a comma, e.g.:
+
+[source,bash,subs=attributes+]
+----
+./mvnw package -DskipTests -Pnative -Dquarkus.native.container-build=true \
+    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:{mandrel-flavor} \
+    -Dquarkus.native.additional-build-args=--native-image-info,--verbose
+----
+
+[TIP]
+====
+Remember that if an argument for `-Dquarkus.native.additional-build-args` includes the `,` symbol,
+it needs to be escaped to be processed correcly, e.g. `\\,`.
+====
+
+Given a native executable, we can use several linux tools to gather information.
```

Comment:
Should this be in a separate subsection?

Reply:
At a first glance I don't think so. This first section focuses on basic things we can do to gather information about the native binary build process and the native binary itself. Hence it's call first debugging steps.